packages/app/src/components/Web3/Web3Modal.tsx

```diff
@@ -1,53 +1,127 @@
-// import * as React from 'react';
-// import { View } from 'react-native';
-
-// import Modal from 'react-native-modal';
-// import QRCode from 'react-native-qrcode-svg';
-
-// function useWeb3Modal() {
-//      function triggerCloseAnimation(): void {
-//         const doc = getDocumentOrThrow();
-//         const modal = doc.getElementById(WALLETCONNECT_MODAL_ID);
-//         if (modal) {
-//             modal.className = modal.className.replace('fadeIn', 'fadeOut');
-//             setTimeout(() => {
-//                 const wrapper = doc.getElementById(WALLETCONNECT_WRAPPER_ID);
-//                 if (wrapper) {
-//                     doc.body.removeChild(wrapper);
-//                 }
-//             }, ANIMATION_DURATION);
-//         }
-//     }
-
-//     function getWrappedCallback(cb: any): any {
-//         return () => {
-//             triggerCloseAnimation();
-//             if (cb) {
-//                 cb();
-//             }
-//         };
-
-//          function open(
-//             uri: string,
-//             cb: any,
-//             qrcodeModalOptions?: IQRCodeModalOptions
-//         ) {
-//             return (
-//                 <Modal
-//                     onModalHide={getWrappedCallback(cb)}
-//                     qrcodeModalOptions={qrcodeModalOptions}
-//                 >
-//                     <View>
-//                         <QRCode size={100} value={uri} />
-//                     </View>
-//                 </Modal>
-//             );
-//         }
-
-//         export function close() {
-//             triggerCloseAnimation();
-//         }
-//     }
-
-//     return { open, close };
-// }
+import React, { useState, useContext, ReactElement } from 'react';
+
+import { Platform, Text, View, StyleSheet, Button, Image, TouchableWithoutFeedback, ImageProps } from 'react-native';
+import { Web3Context } from '../../contexts';
+
+import QRCode from 'react-native-qrcode-svg';
+import Modal from 'react-native-modal';
+import wallets from '../../registry';
+import wcLogo from '../../images/walletconnect-logo.png';
+
+const WalletsGrid = (): ReactElement => (
+  <View style={styles.grid}>  
+    {wallets.map(({ name, logo }: { 
+      name: string, 
+      logo: ImageProps['source']
+    }) => (
+      <View style={styles.item} key={name}>
+        <Image source={logo} style={styles.logo}/>
+        <Text>{name}</Text>
+      </View>
+    ))}
+  </View>
+)
+
+const ConnectButton = (): ReactElement => (
+  <Button 
+    title="Connect" 
+    onPress={() => console.log('connect')}
```

Comment:
This should be a <Linking/> that has the uri. 
https://reactnative.dev/docs/linking